### ISSUE TITLE
Peer tries to roll back protocol version to 17.0, Error while processing ID

### ISSUE THREAD

**Issue by greyghost509** [gsliepen/tinc#455]
I have upgraded/reinstalled one of my tinc instances. Unfortunately, it appears that other machines cannot connect to it anymore although (to my best knowledge) I kept the configuration.

On the machines that try to connect to the upgraded machine (both running 1.1pre18), I get the following error messages:

Apr 17 11:05:18 tinc.l15[27046]: Peer l15_firefly12 (10.117.15.243 port 40893) tries to roll back protocol version to 17.0
Apr 17 11:05:18 tinc.l15[27046]: Error while processing ID from l15_firefly12 (10.117.15.243 port 40893)

I have checked that the keys are correct (so they should not cause an ID error). Any idea what is causing the problem here?

**Comment by f-hollow** [comment]
I ran into the same error. In my case, some machines were running v1.1pre18, other ones were running v1.0.36. Apparently, there were some issues using the experimental Ed25519 key for authentication.

To get around this error, I added to `tinc.conf` of all my v1.1pre18 nodes the line `ExperimentalProtocol = no`. It is also mentioned [here](https://www.tinc-vpn.org/pipermail/tinc/2018-June/005269.html).

**Comment by greyghost509** [comment]
Yes, this solution is working for me, too (either turning off experimental or removing the key).
However, I cannot really see why this should be a "feature". This should be mentioned more clearly in the documentation, and there should be a more explicit error message.
